src/api/app/controllers/source_controller.rb

```diff
@@ -475,141 +474,98 @@ class ProjectReadAccessFailure < APIException
 
   # PUT /source/:project/_meta
   def update_project_meta
-    # init
-    #--------------------
     project_name = params[:project]
     params[:user] = User.current.login
-
-    # init
-    # assemble path for backend
-    path = request.path_info
-    path += build_query_from_hash(params, [:user, :comment, :rev])
-    #allowed = false
     request_data = request.raw_post
 
     # permission check
-    rdata = Xmlhash.parse(request_data)
-    if rdata['name'] != project_name
-      raise ProjectNameMismatch.new "project name in xml data ('#{rdata['name']}) does not match resource path component ('#{project_name}')"
+    request_data = Xmlhash.parse(request_data)
+    if request_data['name'] != project_name
+      raise ProjectNameMismatch, "project name in xml data ('#{request_data['name']}) does not match resource path component ('#{project_name}')"
     end
+
     begin
-      prj = Project.get_by_name rdata['name']
+      project = Project.get_by_name(request_data['name'])
     rescue Project::UnknownObjectError
-      prj = nil
+      project = nil
     end
 
     # projects using remote resources must be edited by the admin
     unless User.current.is_admin?
```

Comment:
I would move this block up to line 486. Then we have 2 logical groups: parsing the xml and processing it and fetching the project and processing that.